addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -1405,7 +1414,11 @@ export function makeActionManager(env, router = _router) {
         // in case an effect is returned from python and there is already an effect
         // attribute on the button, the priority is given to the button attribute
         const effect = params.effect ? evaluateExpr(params.effect) : action.effect;
-        const options = { onClose: params.onClose };
+        const options = {
+            onClose: params.onClose,
+            stackPosition: params.stackPosition,
+            viewType: params.viewType,
+        };
```

Comment:
```suggestion
        const { onClose, stackPosition, viewType } = params;
        const options = { onClose, stackPosition, viewType };
```
I would have done that but I don't know if it is better